pkg/controller/containerjfr/containerjfr_controller_test.go

```diff
@@ -0,0 +1,462 @@
+}
+
+func expectExporterService(client client.Client, controller containerjfr.ReconcileContainerJFR, minimal bool) {
+	service := &corev1.Service{}
+	err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, service)
+	Expect(err).To(HaveOccurred())
+
+	reconcileFully(client, controller, minimal)
+
+	err = client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, service)
+	Expect(err).ToNot(HaveOccurred())
+
+	expectedService := resource_definitions.NewExporterService(test.NewContainerJFR())
+	checkMetadata(&service.ObjectMeta, &expectedService.ObjectMeta)
+	Expect(service.Spec.Type).To(Equal(expectedService.Spec.Type))
+	Expect(service.Spec.Selector).To(Equal(expectedService.Spec.Selector))
+	Expect(service.Spec.Ports).To(Equal(expectedService.Spec.Ports))
+}
+
+func expectCommandChannel(client client.Client, controller containerjfr.ReconcileContainerJFR, minimal bool) {
+	service := &corev1.Service{}
+	err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr-command", Namespace: "default"}, service)
+	Expect(err).To(HaveOccurred())
+
+	reconcileFully(client, controller, minimal)
+
+	err = client.Get(context.Background(), types.NamespacedName{Name: "containerjfr-command", Namespace: "default"}, service)
+	Expect(err).ToNot(HaveOccurred())
+
+	expectedService := resource_definitions.NewCommandChannelService(test.NewContainerJFR())
+	checkMetadata(&service.ObjectMeta, &expectedService.ObjectMeta)
+	Expect(service.Spec.Type).To(Equal(expectedService.Spec.Type))
+	Expect(service.Spec.Selector).To(Equal(expectedService.Spec.Selector))
+	Expect(service.Spec.Ports).To(Equal(expectedService.Spec.Ports))
+}
+
+func expectDeployment(client client.Client, controller containerjfr.ReconcileContainerJFR, minimal bool) {
+	deployment := appsv1.Deployment{}
+	err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, &deployment)
+	Expect(err).To(HaveOccurred())
+
+	reconcileFully(client, controller, minimal)
+
+	err = client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, &deployment)
+	Expect(err).ToNot(HaveOccurred())
+
+	testSpecs := newFakeServiceSpecs(minimal)
+	testTLSConfig := newFakeTLSConfig()
+	testContainer := &rhjmcv1beta1.ContainerJFR{}
+	if minimal {
+		testContainer = test.NewMinimalContainerJFR()
+	} else {
+		testContainer = test.NewContainerJFR()
+	}
+	expectedDeployment := resource_definitions.NewDeploymentForCR(testContainer, &testSpecs, &testTLSConfig)
+	Expect(deployment.ObjectMeta.Name).To(Equal(expectedDeployment.ObjectMeta.Name))
+	Expect(deployment.ObjectMeta.Namespace).To(Equal(expectedDeployment.ObjectMeta.Namespace))
+	Expect(deployment.ObjectMeta.Labels).To(Equal(expectedDeployment.ObjectMeta.Labels))
+	Expect(deployment.ObjectMeta.Annotations).To(Equal(expectedDeployment.ObjectMeta.Annotations))
```

Comment:
This could be done with `checkMetadata` as well, I believe.

Reply:
Yes, I missed that, I fix it!